framework/src/geomsearch/PenetrationLocator.C

```diff
@@ -121,6 +122,38 @@ PenetrationLocator::detectPenetration()
 
   Threads::parallel_reduce(slave_node_range, pt);
 
+  std::vector<dof_id_type> recheck_slave_nodes = pt._recheck_slave_nodes;
+
+  if (recheck_slave_nodes.size() > 0 && _automatic_patch_update)
+  {
+    // Update the patch for this subset of slave nodes and calculate the nearest neighbor_nodes
+    _nearest_node.updatePatch(recheck_slave_nodes);
+
+    // Re-run the penetration thread to see if these nodes are in contact with the updated patch
+    NodeIdRange recheck_slave_node_range(recheck_slave_nodes.begin(), recheck_slave_nodes.end(), 1);
+
+    PenetrationThread pt(_subproblem,
```

Comment:
`pt` here is a shadow variable (an object with the same name as one that was created outside this block), which should be avoided because it's confusing. I think you can actually just re-use the `pt` object created outside this block (line 104), which would cut down on the complexity of this code.

Reply:
Thanks for the comment. I will make this change.